app/components/SavedContracts/__tests__/SavedContract.spec.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+
+import states from '../../../states'
+import SavedContracts from '../SavedContracts'
+
+jest.mock('electron', () => ({
+  app: {
+    getPath() { return 'test' },
+  },
+  ipcRenderer: { send: jest.fn(), on: jest.fn() },
+}))
+
+describe('SavedContracts', () => {
+  describe('sortSavedContractsByExpiry', () => {
+    const UnWrappedComponent =
+      SavedContracts.wrappedComponent.wrappedComponent.wrappedComponent.wrappedComponent
+    const component = shallow(<UnWrappedComponent {...states} />)
+    const sortFunction = component.instance().sortSavedContractsByExpiry
+    describe('when contracts contain data with undefined expiry', () => {
```

Comment:
This test is for the sort function not the delete button. Indeed the sort function is a pure function and does not need to live in the component. The existing implementation of the sort was not correctly sorting the contracts and since I practice TDD a lot, I tried to test this with minimal refactor of the existing component. Of course I could have move things around even more - but it becomes a bit tangential to the work at hand and I think there is a bigger discussion to be had regarding testing philosophy as well as component structures that facilitate the testing philosophy the project intends to follow.

I do think reviewing the test pyramid in relation to this project should be done (post genesis) and sign posting the outcomes of such discussions in the dev readme or something similar would be valuable.